include/deal.II/meshworker/scratch_data.h

```diff
@@ -0,0 +1,1264 @@
+     */
+    /**@{*/
+
+    /**
+     * Initialize the internal FEValues with the given @p cell, and return
+     * a reference to it.
+     *
+     * After calling this function, get_current_fe_values() will return the
+     * same object of this method, as an FEValuesBase reference.
+     */
+    const FEValues<dim, spacedim> &
+    reinit(
+      const typename DoFHandler<dim, spacedim>::active_cell_iterator &cell);
+
+    /**
+     * Initialize the internal FEFaceValues to use the given @p face_no on the given
+     * @p cell, and return a reference to it.
+     *
+     * After calling this function, get_current_fe_values() will return the
+     * same object of this method, as an FEValuesBase reference.
+     */
+    const FEFaceValues<dim, spacedim> &
+    reinit(const typename DoFHandler<dim, spacedim>::active_cell_iterator &cell,
+           const unsigned int face_no);
+
+    /**
+     * Initialize the internal FESubfaceValues to use the given @p subface_no,
+     * on @p face_no, on the given @p cell, and return a reference to it.
+     *
+     * After calling this function, get_current_fe_values() will return the
+     * same object of this method, as an FEValuesBase reference.
+     *
+     * If @p subface_no is numbers::invalid_unsigned_int, the reinit() function
+     * that takes only the @p cell and the @p face_no is called.
+     */
+    const FEFaceValuesBase<dim, spacedim> &
+    reinit(const typename DoFHandler<dim, spacedim>::active_cell_iterator &cell,
+           const unsigned int face_no,
+           const unsigned int subface_no);
+
+    /**
+     * Get the currently initialized FEValues.
+     *
+     * This function will return the internal FEValues if the
+     * reinit(cell) function was called last. If the reinit(cell, face_no)
+     * function was called, then this function returns the internal
+     * FEFaceValues, and if the reinit(cell, face_no, subface_no) function was
+     * called (with a valid @p subface_no argument), it returns the internal
+     * FESubfaceValues object.
+     */
+    const FEValuesBase<dim, spacedim> &
+    get_current_fe_values() const;
+
+    /**
+     * Return the quadrature points of the internal FEValues object.
+     */
+    const std::vector<Point<spacedim>> &
+    get_quadrature_points() const;
+
+    /**
+     * Return the JxW values of the internal FEValues object.
+     */
+    const std::vector<double> &
+    get_JxW_values() const;
+
+    /**
+     * Return the last computed normal vectors.
+     */
+    const std::vector<Tensor<1, spacedim>> &
+    get_normal_vectors() const;
+
+    /**
+     * Return the local dof indices for the cell passed the last time the
+     * reinit() function was called.
+     */
+    const std::vector<types::global_dof_index> &
+    get_local_dof_indices() const;
+
+    /** @} */ // CurrentCellMethods
+
+    /**
+     * @defgroup NeighborCellMethods Methods to work on neighbor cell
```

Comment:
same here and below

Reply:
https://www.star.bnl.gov/public/comp/sofi/doxygen/grouping.html

Reply:
I think you misunderstand what `@defgroup` does. It creates a group *globally* across the whole project. We do this, for example, for "linear algebra classes", "finite element classes", etc in `doc/doxygen/headers/*`.

But you really only want to group a bunch of member functions together, and the name of the group should not propagate beyond the class you're documenting. That's what `@name` is there for. (Take a look at the `Triangulation` class, for example, to see how we use it.)